src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -755,6 +761,29 @@ export const GlobalProvider = memo(
             [modifyNode, schemata]
         );
 
+        const useInputSize = useCallback(
+            (
+                id: string,
+                inputId: InputId,
+                inputSize?: InputSize
+            ): readonly [Size, (size: Size) => void, () => void] => {
+                const currentSize = inputSize ? inputSize[inputId] : { width: 0, height: 0 };
+                const setInputSize = (size: Size) => {
+                    modifyNode(id, (old) => {
+                        const nodeCopy = copyNode(old);
+                        nodeCopy.data.inputSize = {
+                            ...nodeCopy.data.inputData,
+                            [inputId]: size,
+                        };
+                        return nodeCopy;
+                    });
+                };
+                const resetInputSize = () => setInputSize({ width: 0, height: 0 } as Size);
```

Comment:
This type cast is unnecessary.

Also, please try to avoid type casts in general. A type cast is intended to be used can't correctly determine the type of something. Since you essentially say "trust me on that one", all safety TS provides is thrown out of the window. It's the same with non-nullish assertions. They are essentially hacks.

```suggestion
                const resetInputSize = () => setInputSize({ width: 0, height: 0 });
```